ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/DataSourceExtractor.java

```diff
@@ -131,6 +132,35 @@ public Void visitAliasedRelation(final AliasedRelation relation, final Void ctx)
       return null;
     }
 
+    private String checkAlternatives(final SourceName sourceName) {
+      String hint = "";
+
+      final Set<SourceName> existingSources = metaStore.getAllDataSources().keySet();
+      for (SourceName name:existingSources) {
+        if (name.text().equalsIgnoreCase(sourceName.text())) {
+          /* if source name in metastore contains small letters, the query source name must be
+          / wrapped in double quotes with the correct order of small and capital letters */
+          if (!name.text().toUpperCase().equals(name.text())
+              && sourceName.text().toUpperCase().equals(sourceName.text())) {
+            hint = String.format(
+                "\nHint: try wrapping the source name in double quotes "
+                    + "with the correct order of small and capital letters: \"" + name.text() + "\""
+            );
+          /* if the source name passed by the query is wrapped in double quotes,
+          the double quotes must be removed, or it must have the same order of small and capital
+          letters as the created source */
+          } else if (!sourceName.text().toUpperCase().equals(sourceName.text())) {
+            hint = String.format(
+                "\nHint: try removing double quotes from the source name or query the source name "
```

Comment:
This hint is not correct. For example: 

```
ksql> CREATE STREAM "bAr" (x STRING KEY) WITH (kafka_topic='s1', format='json', partitions=1);
ksql> select * from "bAR";
Exception while preparing statement: bAR does not exist.
Hint: try removing double quotes from the source name or query the source name with the correct order of small and capital letters: bAr
Statement: select * from "bAR";
Caused by: bAR does not exist.
Hint: try removing double quotes from the source
	name or query the source name with the correct order of small and capital
	letters: bAr
ksql> select * from bAR;
Exception while preparing statement: BAR does not exist.
Hint: try wrapping the source name in double quotes with the correct order of small and capital letters: "bAr"
Statement: select * from bAR;
Caused by: BAR does not exist.
Hint: try wrapping the source name in double
	quotes with the correct order of small and capital letters: "bAr"
```

I think we should tackle this issue in one of the two ways. Either we speculatively try for most frequent typos and a map.get for them, like `getAllDataSources.get(sourceName.lowerCase())` , or we do an exhaustive search, but then the hint must be precise.

Reply:
Even though your approach is considerable, I think the hint is 100% correct, since it encourages the user to query the source name  `with the correct order of small and capital letters: bAr `.  More over it shows the correct spelling as `bAr`

Reply:
Can you please elaborate? Maybe I am missing something:

```sql
ksql> CREATE STREAM "bAr" (x STRING KEY) WITH (kafka_topic='s1', format='json', partitions=1);
ksql> select * from "bAR";
Exception while preparing statement: bAR does not exist.
Hint: try removing double quotes from the source name or query the source name with the correct order of small and capital letters: bAr
Statement: select * from "bAR";
Caused by: bAR does not exist.
Hint: try removing double quotes from the source
	name or query the source name with the correct order of small and capital
	letters: bAr
```

This hint suggests removing double quotes. This is not a good advice as we do need the double quotes to run the query correctly.

In other words, the hint suggests to change the query in the following way

```sql
ksql> select * from bAr;
Exception while preparing statement: BAR does not exist.
Hint: try wrapping the source name in double quotes with the correct order of small and capital letters: "bAr"
Statement: select * from bAr;
Caused by: BAR does not exist.
Hint: try wrapping the source name in double
	quotes with the correct order of small and capital letters: "bAr"
```

which is not correct. The correct query would be with both the right spelling and double quotes:

```sql
ksql> select * from "bAr";
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|X                                                                                                                                                                                           |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
Query Completed
Query terminated
```

Reply:
> Can you please elaborate? Maybe I am missing something:

Since there are two suggestions combined with **OR**, only the second part which is about correcting the order of capital/small letters must be done. 
Anyway, if the message seems misleading to YOU, it may also be inappropriate for the end user!